pkg/controller/gittrack/gittrack_controller.go

```diff
@@ -537,16 +537,20 @@ func (r *ReconcileGitTrack) Reconcile(request reconcile.Request) (reconcile.Resu
 	r.recorder.Eventf(instance, apiv1.EventTypeNormal, "CheckoutSuccessful", "Successfully checked out '%s' at '%s'", instance.Spec.Repository, instance.Spec.Reference)
 
 	// Attempt to parse k8s objects from files
-	objects, errors := objectsFrom(files)
-	if len(errors) > 0 {
-		sOpts.parseError = fmt.Errorf(strings.Join(errors, ",\n"))
+	objects, fileErrors := objectsFrom(files)
+
+	// Update status with the number of objects discovered
+	sOpts.discovered = int64(len(objects))
+
+	if len(fileErrors) > 0 {
+		sOpts.discovered += int64(len(fileErrors))
+		sOpts.ignored += int64(len(fileErrors))
```

Comment:
Does not seem to work. Why?

Reply:
How are you testing this? Do we have a test case? If not, could you try and devise a test case for this?

I think you can use a script to mount the test repository, commit a broken file and copy the sha of the commit, then create a test that checks out the broken sha. I think @mthssdrbrg wrote some instructions somewhere 🤔

Reply:
I don't know if there are any instructions apart from the [script itself](https://github.com/pusher/faros/blob/master/hack/update-fixtures-repo) though :/

Reply:
Instructions would be something along this:

1. Run `hack/update-fixtures-repo`, which will drop you into a shell, in a temporary directory with the contents of `pkg/controller/gittrack/fixtures/repo.tgz` extracted.
2. Add new files/modify present ones/etc and make a new commit (and note the SHA of the new commit).
3. Exit the shell with `exit 0` (or `exit 1` if you need to discard your changes).
4. Write tests using the new SHA.
5. Commit the tests and the updated `repo.tgz`.
6. ???
7. Profit

I should probably make a PR that includes them :sweat_smile: